repository/activity_repository/activity_repository.go

```diff
@@ -0,0 +1,40 @@
+package activity_repository
+
+import (
+	"encoding/json"
+	"gin/config"
+	"gin/models"
+	"gopkg.in/redis.v5"
+)
+
+func (rs RedisActivityRepository) GetUserActivity(username string) ([]string, error) {
+	return rs.client.LRange(username, 0, 2).Result()
+}
+
+func (rs RedisActivityRepository) SetUserActivity(username string, request models.UserRequest) error {
+	reqJson, err := json.Marshal(request)
+	if err != nil {
+		return err
+	}
+
+	cmd := rs.client.LPush(username, reqJson)
+	if err := cmd.Err(); err != nil {
+		return err
+	}
+
+	err = rs.client.LTrim(username, 0, 2).Err()
+	return err
+}
+
+type ActivityService interface {
```

Comment:
Put the interface declaration in a different file from the implementation.

Reply:
We usually name interfaces with the suffix "er" in go. like "cacher", "logger"...